### DISCUSSION TITLE
Layout calculation

### DISCUSSION THREAD

**Discussion by willeke1234** [GrandOrgue/grandorgue#1290] [Developers]
In `GOGUILayoutEngine::Update`, the calculation of the y's starts at the bottom and the height of the panel areas is subtracted. `GetPedalHeight()` is subtracted twice. Shouldn't [`m_CenterY -= m_metrics.GetPedalHeight();`](https://github.com/GrandOrgue/grandorgue/blob/master/src/grandorgue/gui/GOGUILayoutEngine.cpp#L247) be `m_CenterY -= m_metrics.GetButtonHeight();`?

**Comment by oleg68** [comment]
@willeke1234 I tried to understand how this code works, but if I remove one of two substraction of `m_metrics.GetPedalHeight()`, the main organ panel looks incorrectly.

**Comment by willeke1234** [comment]
I think the second `GetPedalHeight()` is subtracted for the pistons and should be `GetButtonHeight()` instead. Both are 40 by default and not present in the sample sets I tried. Either I don't understand how it works or it's a bug.

By the way, I was looking for an image of the layout metrics. Does it exist?

**Reply by oleg68** [reply]
> By the way, I was looking for an image of the layout metrics. Does it exist?

GrandOrgue Demo Organ.

**Comment by willeke1234** [comment]
When I add `DispPedalHeight=80` to the demo organ then an extra piece of vertical wood pattern is added.

![image](https://user-images.githubusercontent.com/97431382/205942016-d5906b74-754d-4262-b874-2067188faf85.png)

I think `DispPedalHeight` isn't used in panels.
